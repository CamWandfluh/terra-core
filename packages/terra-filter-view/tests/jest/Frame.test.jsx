import React from 'react';
import Frame from '../../src/_Frame';
import intlContexts from './intl-context-setup';

describe('Frame', () => {
  it('should render a dropdown variant', () => {
    const wrapper = shallow(<Frame variant="dropdown" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list variant', () => {
    const wrapper = shallow(<Frame variant="list" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a dropdown variant with a placeholder', () => {
    const wrapper = shallow(<Frame variant="dropdown" placeholder="Placeholder" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list variant with a placeholder', () => {
    const wrapper = shallow(<Frame variant="list" placeholder="Placeholder" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a disabled dropdown variant', () => {
    const wrapper = shallow(<Frame variant="dropdown" disabled />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a disabled list variant', () => {
    const wrapper = shallow(<Frame variant="list" disabled />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });
});
