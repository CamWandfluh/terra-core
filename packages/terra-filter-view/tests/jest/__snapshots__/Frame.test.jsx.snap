// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame should render a disabled dropdown variant 1`] = `
<div
  className="select-parent"
>
  <div
    aria-disabled={true}
    aria-expanded={false}
    aria-haspopup="true"
    className="select dropdown is-disabled"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="display"
      onMouseDown={[Function]}
    >
      <input
        aria-label="search"
        className="search-input"
        disabled={true}
        onChange={[Function]}
        onMouseDown={[Function]}
      />
    </div>
    <Button
      className="button"
      icon={
        <IconSearch
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onMouseDown={[Function]}
      text="Search"
      type="button"
      variant="neutral"
    />
  </div>
</div>
`;

exports[`Frame should render a disabled list variant 1`] = `
<div
  className="select-parent"
>
  <div
    aria-disabled={true}
    aria-expanded={false}
    className="select list is-disabled"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="display"
      onMouseDown={[Function]}
    >
      <input
        aria-label="search"
        className="search-input"
        disabled={true}
        onChange={[Function]}
        onMouseDown={[Function]}
      />
    </div>
    <Button
      className="button"
      icon={
        <IconSearch
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onMouseDown={[Function]}
      text="Search"
      type="button"
      variant="neutral"
    />
  </div>
  <div
    className="box"
    id="terra-filter-box"
  />
</div>
`;

exports[`Frame should render a dropdown variant 1`] = `
<div
  className="select-parent"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    aria-haspopup="true"
    className="select dropdown"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="display"
      onMouseDown={[Function]}
    >
      <input
        aria-label="search"
        className="search-input"
        disabled={false}
        onChange={[Function]}
        onMouseDown={[Function]}
      />
    </div>
    <Button
      className="button"
      icon={
        <IconSearch
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onMouseDown={[Function]}
      text="Search"
      type="button"
      variant="neutral"
    />
  </div>
</div>
`;

exports[`Frame should render a dropdown variant with a placeholder 1`] = `
<div
  className="select-parent"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    aria-haspopup="true"
    className="select dropdown"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="display"
      onMouseDown={[Function]}
    >
      <input
        aria-label="search"
        className="search-input"
        disabled={false}
        onChange={[Function]}
        onMouseDown={[Function]}
        placeholder="Placeholder"
      />
    </div>
    <Button
      className="button"
      icon={
        <IconSearch
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onMouseDown={[Function]}
      text="Search"
      type="button"
      variant="neutral"
    />
  </div>
</div>
`;

exports[`Frame should render a list variant 1`] = `
<div
  className="select-parent"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="select list"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="display"
      onMouseDown={[Function]}
    >
      <input
        aria-label="search"
        className="search-input"
        disabled={false}
        onChange={[Function]}
        onMouseDown={[Function]}
      />
    </div>
    <Button
      className="button"
      icon={
        <IconSearch
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onMouseDown={[Function]}
      text="Search"
      type="button"
      variant="neutral"
    />
  </div>
  <div
    className="box"
    id="terra-filter-box"
  />
</div>
`;

exports[`Frame should render a list variant with a placeholder 1`] = `
<div
  className="select-parent"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="select list"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="display"
      onMouseDown={[Function]}
    >
      <input
        aria-label="search"
        className="search-input"
        disabled={false}
        onChange={[Function]}
        onMouseDown={[Function]}
        placeholder="Placeholder"
      />
    </div>
    <Button
      className="button"
      icon={
        <IconSearch
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onMouseDown={[Function]}
      text="Search"
      type="button"
      variant="neutral"
    />
  </div>
  <div
    className="box"
    id="terra-filter-box"
  />
</div>
`;
